### PR TITLE
Semi auto derivation

### PR DESCRIPTION
**Pull request by mbore**

### CODE REVIEW

File: src/core/macro.scala [hunk 1]
```diff
@@ -12,6 +12,12 @@ object Macro:
   inline def paramTypeAnns[T]: List[(String, List[Any])] = ${paramTypeAnns[T]}
   inline def repeated[T]: List[(String, Boolean)] = ${repeated[T]}
   inline def typeInfo[T]: TypeInfo = ${typeInfo[T]}
+  inline def summonOption[T]: Option[T] = ${summonOptionImpl[T]}
+
+  def summonOptionImpl[T: Type](using Quotes): Expr[Option[T]] =
+    Expr.summon[T] match
```
Comment (adamw) [inline src/core/macro.scala:18]: can't this simply be `Expr.summon[T]` without the match? :)

**Comment by adamw** [comment]
Maybe we could add the recursive tests here as well?

+ some info in the readme about semi/auto derivation would be nice as well

File: src/core/macro.scala (hunk 1 above)
Comment (KacperFKorban) [reply src/core/macro.scala:18]: The generics are switched. `Expr[Option[T]]` vs `Option[Expr[T]]`

Comment (adamw) [reply src/core/macro.scala:18]: ah 🤦

**Comment by adamw** [comment]
Thanks! :)

**Pull request merged**